include/autoppl/algorithm/mh.hpp

```diff
@@ -64,7 +65,13 @@ inline void mh__(ModelType& model,
     std::uniform_real_distribution unif_sampler(0., 1.);
 
     for (size_t iter = 0; iter < n_sample + warmup; ++iter) {
-        
+        if (iter % ((n_sample + warmup) / 100) == 0) {
```

Comment:
Seems like we can abstract this out into a logging function - can we actually start another header file in autoppl/logger dir with such function and reuse this function in NUTS as well

Reply:
Ok makes sense. How about this?